azurerm/resource_arm_kubernetes_cluster_auth_test.go

```diff
@@ -0,0 +1,345 @@
+					resource.TestCheckResourceAttr(resourceName, "role_based_access_control.0.enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "role_based_access_control.0.azure_active_directory.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "role_based_access_control.0.azure_active_directory.0.client_app_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "role_based_access_control.0.azure_active_directory.0.server_app_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "role_based_access_control.0.azure_active_directory.0.server_app_secret"),
+					resource.TestCheckResourceAttrSet(resourceName, "role_based_access_control.0.azure_active_directory.0.tenant_id"),
+					resource.TestCheckResourceAttr(resourceName, "kube_admin_config.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "kube_admin_config_raw"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"service_principal.0.client_secret",
+					"role_based_access_control.0.azure_active_directory.0.server_app_secret",
+				},
+			},
+			{
+				// should be no changes since the default for Tenant ID comes from the Provider block
+				Config:   testAccAzureRMKubernetesCluster_roleBasedAccessControlAAD(ri, location, clientId, clientSecret, tenantId),
+				PlanOnly: true,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKubernetesClusterExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"service_principal.0.client_secret",
+					"role_based_access_control.0.azure_active_directory.0.server_app_secret",
+				},
+			},
+		},
+	})
+}
+
+func testAccAzureRMKubernetesCluster_apiServerAuthorizedIPRanges(rInt int, clientId string, clientSecret string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestvirtnet%d"
+  address_space       = ["10.1.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "acctestsubnet%d"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefix       = "10.1.0.0/24"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+
+  default_node_pool {
+    name           = "default"
+    node_count     = 1
+    vm_size        = "Standard_DS2_v2"
+    vnet_subnet_id = azurerm_subnet.test.id
```

Comment:
@tombuildsstuff – I was wondering, would it be worth surfacing the fact that the `vnet_subnet_id` on the `default_node_pool` is also the subnet that the entire cluster gets launched into?

```shell
az aks create -h | grep vnet-subnet -A1
    --vnet-subnet-id                          : The ID of a subnet in an existing VNet into which to
                                                deploy the cluster.
```

https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-create

Reply:
FWIW we plan to support assigning a separate subnet per agentpool, but haven't implemented it quite yet. Hence today the subnet property on additional agentpools doesn't do much yet.

Reply:
Oh interesting. I hadn't even considered that.

Reply:
I agree with you surfacing that info is helpful, just want to make sure it's easily updated when that new func is available :)

Reply:
@jluk just to ensure I've got this right - from testing IIRC at this time all of the Subnet ID's need to be set to the same value for the meantime; rather than just being defined on the top level resource?

Reply:
Correct today all of the subnetIds need to be the same.

Reply:
When creating a new agent pool, if the subnetid is not provided, the default will be inferred from the primary node pool's subnetid. So you can either input the same id across all of them or not provide a value for it to be inferred, any other inputs will receive an error.

Reply:
@jluk thanks for confirming - will get that updated 👍

Reply:
PR: https://github.com/terraform-providers/terraform-provider-azurerm/pull/4946